WalletWasabi/Services/WalletService.cs

```diff
@@ -71,7 +62,8 @@ public static bool DownloadingBlock
 
 		public ConcurrentDictionary<uint256, (Height height, DateTimeOffset dateTime)> ProcessedBlocks { get; }
 
-		public ObservableConcurrentHashSet<SmartCoin> Coins { get; }
+		public ICoinsView Coins { get; }
```

Comment:
Better naming? Which Coins, Unspent coins?

Reply:
Very good question. There is only one place in all the code base where we need the `spent` txos and it is for displaying the history of transactions. In the rest of the code we always use the **utxo** (coins). In this PR `Coins` contains the list of utxos (also known as coins).

Reply:
Before I could always see that we are filtering the unspent Coins - so for this I was always sure that they were the UTXO-s. Now I just saw Coins in the code and it was confusing. This is a better solution obviously. Could you decorate the code with some comments?